tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,143 @@
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach (keys %virt_autotest::common::guests) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ($sys_disk, $dev, $lvm_disk);
+    # Get current system disk name
+    $sys_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /)';
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    # Use a additional free hard disk for LVM volumes
+    $dev = "/dev/";
+    if ($sys_disk eq $disks[0]) {
+        $lvm_disk = $dev . $disks[2];
+    } else {
+        $lvm_disk = $dev . $disks[0];
+    }
+    record_info "Assign a New Disk:", "$lvm_disk";
+    # Wipe Hard Disk Clean via dd for assigned a new full disk
+    wipe_hard_disk($lvm_disk);
+    # Create new disk partition for LVM volumes
+    assert_script_run 'echo -e "g\nn\n\n\n+20G\nt\n8e\np\nw" | fdisk ' . $lvm_disk;
+    assert_script_run 'lsblk';
+    # Create a Physical Volume (PV) with LVM
+    my $timeout = 180;
+    record_info "Create a Physical Volume";
+    validate_script_output("pvcreate ${lvm_disk}1", sub { m/successfully created/ }, $timeout);
+    validate_script_output("pvdisplay", sub { m/${lvm_disk}1/ }, $timeout);
+    # Create a Volume Group (VG) with LVM named 'lvm_vg'
+    record_info "Create a Volume Group";
+    validate_script_output("vgcreate lvm_vg ${lvm_disk}1", sub { m/successfully created/ }, $timeout);
+    validate_script_output("vgdisplay lvm_vg", sub { m/lvm_vg/ }, $timeout);
+
+    ## About LVM volume group storage pool management
+    # Use an LVM Volume Group (VG) as a storage pool named 'guest_image_lvm'
+    record_info "LVM Storage Pool define";
+    assert_script_run "virsh pool-define-as guest_image_lvm logical --source-name lvm_vg --target /dev/lvm_vg";
+
+    record_info "LVM Storage Pool list";
+    assert_script_run "virsh pool-list --all | grep guest_image_lvm";
+
+    record_info "LVM Storage Pool start";
+    assert_script_run "virsh pool-start guest_image_lvm";
+
+    record_info "LVM Storage Pool autostart";
+    assert_script_run "virsh pool-autostart guest_image_lvm";
+
+    record_info "LVM Storage Pool info";
+    assert_script_run "virsh pool-info guest_image_lvm";
+
+    # Create a Storage Volume from LVM storage pool
+    record_info "Create a LVM Storage Volume";
+    assert_script_run("virsh vol-create-as guest_image_lvm $_-storage 1G", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "List LVM Storage Volumes";
+    assert_script_run("ls /dev/lvm_vg", 120) foreach (keys %virt_autotest::common::guests);
```

Comment:
the foreach loop seems to be useless as the ls command is nothing to do with any guest.

Reply:
imho, just confirm each guest system is available to access /dev/lvm_vg. 
had removed this foreach loop.